src/main/groovy/com/avioconsulting/mule/linter/model/pom/PomArtifact.groovy

```diff
@@ -3,7 +3,7 @@ package com.avioconsulting.mule.linter.model.pom
 import com.avioconsulting.mule.linter.parser.MuleXmlParser
 import groovy.xml.slurpersupport.GPathResult
 
-class PomPlugin {
+class PomArtifact {
```

Comment:
A Pom Dependency isn't the same as a Pom Plugin definition.  Are you sure this doesn't change any of the existing pom plugin behavior?  We should differentiate between a Pom Dependency and Pom Plugin definition, although a base class with similar attributes may be appropriate.  There really isn't such a thing as a "Pom Artifact"  so we should probably be specific and the new use case is a pom dependency

Reply:
I did change PomPlugin to PomPlugin where it was being used had couple of places so behavior is same.
It just traverse the xml for plugin: build.plugins.plugin , dependency: dependencies.dependency. This logic is in PomFile in its equivalent method getPlugin and getDependency.

So it had only groupId, artifactId, version attribute. that we checked for plugin. Thought about doing it the way you mentioned but I Wasn't sure what to call the base class, But I can do the way you mentioned.